torchdata/datapipes/iter/util/plain_text_reader.py

```diff
@@ -0,0 +1,237 @@
+        if not self._strip_newline:
+            yield from stream
+            return
+
+        for line in stream:
+            if self._strip_newline:
+                if isinstance(line, str):
+                    yield line.strip("\n")
+                else:
+                    yield line.strip(b"\n")
+            else:
+                yield line
+
+    def decode(self, stream: Iterator[Union[str, bytes]]) -> Iterator[Union[str, bytes]]:
+        if not self._decode:
+            yield from stream
+            return
+
+        for line in stream:
+            if self._decode and isinstance(line, bytes):
+                yield line.decode(self._encoding, self._errors)
+            else:
+                yield line
+
+    def return_path(self, stream: Iterator[D], *, path: str) -> Iterator[Union[D, Tuple[str, D]]]:
+        if not self._return_path:
+            yield from stream
+            return
+
+        for data in stream:
+            if self._return_path:
+                yield path, data
+            else:
+                yield data
+
+
+@functional_datapipe("readlines")
+class LineReaderIterDataPipe(IterDataPipe[Union[Union[str, bytes], Tuple[str, Union[str, bytes]]]]):
+    r"""
+    Iterable DataPipe that accepts a DataPipe consisting of tuples of file name and string data stream,
+    and for each line in the stream, it yields a tuple of file name and the line
+
+    Args:
+        source_datapipe: a DataPipe with tuples of file name and string data stream
+        skip_lines: number of lines to skip at the beginning of each file
+        strip_newline: if True, the new line character will be stripped
+        decode: if True, this will decode the contents of the file based on the specified encoding
+        encoding: the character encoding of the files (default='utf-8')
+        errors: the error handling scheme used while decoding
+        return_path: if True, each line will return a tuple of path and contents, rather
+            than just the contents
+    """
+
+    def __init__(
+        self,
+        source_datapipe,
+        *,
+        skip_lines: int = 0,
+        strip_newline: bool = True,
+        decode: bool = False,
+        encoding="utf-8",
+        errors: str = "ignore",
+        return_path: bool = True,
+    ):
+        self.source_datapipe = source_datapipe
+        self._helper = PlainTextReaderHelper(
+            skip_lines=skip_lines,
+            strip_newline=strip_newline,
+            decode=decode,
+            encoding=encoding,
+            errors=errors,
+            return_path=return_path,
+        )
+
+    def __iter__(self):
+        for path, stream in self.source_datapipe:
+            stream = self._helper.skip_lines(stream)
+            stream = self._helper.strip_newline(stream)
+            stream = self._helper.decode(stream)
+            stream = self._helper.return_path(stream, path=path)
+            yield from stream
+
+
+class _CSVBaseParserIterDataPipe(IterDataPipe):
```

Comment:
I have a feeling to rename these DataPipe all to `Reader` as they become to share same functionalities.

Reply:
Sure, can do. Should this happen here or in a follow-up PR?

Reply:
Separate PR makes more sense.